### DISCUSSION TITLE
milvus cdc/backup integration

### DISCUSSION THREAD

**Discussion by rere950303** [milvus-io/milvus#37450] [Q&A and General discussion]
@SimFG thanks for your works.!

Could I know when the integration of `backup` and `CDC` functionality will be implemented?
It would be great if we could know how far the backup has progressed and have CDC continue from that point, but when making a backup request, the `channel_checkpoints` value is not included in the response.

**Comment by SimFG** [comment]
You can try using the latest branch of the backup and cdc tools, which currently support returning channel checkpoints.

**Reply by rere950303** [reply]
@SimFG 
i am using
- backup: `v0.4.26`
- cdc: `v2.0.0-rc2`

Isn't this the latest version!?

this is my backup log
```
2024.11.05T17:48:48.207[2024/11/05 12:56:29.445 +09:00] [INFO] [core/backup_impl_create_backup.go:540] [[01;31m[KbackupCollectionExecute[m[K] [collectionMeta="id:\"efdbe544-9b29-11ef-a981-1206abf137e1\" start_time:1730778989 collection_id:453712889885696211 db_name:\"default\" collection_name:\"test1_back\" schema:<name:\"test1_back\" fields:<fieldID:100 name:\"pk\" is_primary_key:true data_type:Int64 autoID:true > fields:<fieldID:101 name:\"vector\" data_type:FloatVector type_params:<key:\"dim\" value:\"128\" > > > shards_num:1 consistency_level:Bounded has_index:true index_infos:<field_name:\"vector\" index_name:\"index\" index_type:\"AUTOINDEX\" params:<key:\"index_type\" value:\"AUTOINDEX\" > params:<key:\"metric_type\" value:\"L2\" > > load_state:\"Loaded\" backup_physical_timestamp:1730778989 "]
```

**Reply by rere950303** [reply]
@SimFG 

When using CDC, how should i set the `positions` parameter if we want to replicate data from the initial position of the collection?

**Reply by wayblink** [reply]
@rere950303 What is your milvus version. It looks like milvus-backup didn't get the channel-checkpoint meta from milvus through Flush API

**Reply by SimFG** [reply]
What version of milvus are you using?

**Reply by rere950303** [reply]
@wayblink @SimFG  i am using milvus 2.3.12 versoin

**Reply by wayblink** [reply]
> @wayblink @SimFG i am using milvus 2.3.12 versoin

this function is not supported in v2.3.x. Please try the latest v2.4.x versions

**Reply by rere950303** [reply]
@SimFG @wayblink i upgrade to v2.4.15 but same issue occurs..

![스크린샷 2024-11-06 오후 3 42 25](https://github.com/user-attachments/assets/3f8ee976-cb8d-4f6d-bc30-b6f282550148)

in attu, display previous version, could you let me the reason?

**Reply by SimFG** [reply]
@rere950303 Can you provide the milvus log? I can help you confirm the version.

**Reply by rere950303** [reply]
@SimFG 
Even after a normal upgrade to 2.4.15, the `channel_checkpoints` field does not exist in the backup request response.
```json
{
    "async": false,
    "backup_name": "test",
    "collection_names": [
        "test1_back"
    ]
}
```
```json
{
    "requestId": "cc08a86b-9c1d-11ef-86c6-8edd61a93a34",
    "msg": "success",
    "data": {
        "id": "cc08a86b-9c1d-11ef-86c6-8edd61a93a34",
        "state_code": 2,
        "start_time": 1730883726343,
        "end_time": 1730883734449,
        "progress": 0,
        "name": "dsfsdf",
        "size": 0,
        "milvus_version": "v2.4.15"
    }
}
```

**Reply by SimFG** [reply]
@wayblink can you help checkout it?

**Reply by rere950303** [reply]
@wayblink `channel_checkpoints` exist in logs of cdc, but does not exist in the backup request response.

**Reply by rere950303** [reply]
@wayblink 
Do we originally need to check `channel_checkpoints` information through log files, and is it not possible to check through backup request API responses? I'm asking because I can check it through API responses, so I'm wondering if I configured something incorrectly.

**Reply by rere950303** [reply]
I apologize. I have confirmed that it can be checked through the `get_backup` endpoint.
